File: src/main/java/corp/soyummy/subscribes/SubscribeRepository.java
package corp.soyummy.subscribes;

import org.springframework.stereotype.Repository;
import org.springframework.data.mongodb.repository.MongoRepository;

import org.bson.types.ObjectId;
import java.util.List;

@Repository
public interface SubscribeRepository extends MongoRepository<Subscribe, String> {
    List<Subscribe> findByEmailContainingIgnoreCase(String email);
    boolean existsByOwnerIdAndEmail(ObjectId ownerId, String email);
    void deleteByOwnerIdAndEmail(ObjectId ownerId, String email);
}

File: src/main/kotlin/com/soyummy/subscribes/SubscribeRepository.kt
package com.soyummy.subscribes

import org.springframework.stereotype.Repository
import org.springframework.data.mongodb.repository.MongoRepository
import org.bson.types.ObjectId

@Repository
interface SubscribeRepository: MongoRepository<Subscribe, String> {
    fun findByEmailContainingIgnoreCase(email: String): List<Subscribe>
    fun existsByOwnerIdAndEmail(ownerId: ObjectId, email: String): Boolean
    fun deleteByOwnerIdAndEmail(ownerId: ObjectId, email: String)
}